### PR TITLE
Find leftmost non repeating character in a string #868

### PR DESCRIPTION
**Pull request by salonishah01**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Find leftmost non repeating character in a string #868 
Python

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Find leftmost non repeating character in a string 
#868

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@salonishah01 Instead of just returning `1` and `-1` Display the non-repeated character if it's found and display a proper message when there is no non-repeating character in the given string.

**Comment by salonishah01** [comment]
@Kajol-Kumari Requested changes are done.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@salonishah01 Format your code according to PEP 8 rules. You can refer to this http://pep8online.com/.

**Comment by salonishah01** [comment]
@iamrajiv Code formatted accordingly.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@salonishah01 Update like this
```py
"""
Algorithm to find the first non-repeating character in a string
Input the string from the user.
Start traversing the string using two loops.
Use the first loop to scan the characters of the string one by one.
Use the second loop to find if the current character is occurring
in the latter part of the string or not.
If it is not occurring, print that character.
Else, continue traversing.
"""


def firstNotRepeatingCharacter(string):
    for i in range(len(string)):
        flag = False
        for j in range(len(string)):
            if string[i] == string[j] and i != j:
                flag = True
        if not flag:
            return string[i]
    return "_"

string = input("Enter a string: ")
index = firstNotRepeatingCharacter(string)
if index is None:
    print('No non repeating character found.
          All the characters are repeating.')
else:
    ind = string.find(index)
    print("Non Repeating character " +
          str(string[ind]) + " found at index " + str(ind))

"""
Sample input 1:
Input: "abaacd"
Output: Non Repeating character b found at index 1
Sample input 2:
Input:"aabbccdd"
Output: No non repeating character found. All the characters are repeating.
Time Complexity: O(n*2) - n being the length of string
Space Complexity: O(n)
"""
```

**Review by iamrajiv — APPROVED** [review]

File: Python/other/left_nonRepeating_char.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+"""
+Sample input 1:
+Input: "abaacd"
+Output: Non Repeating character b found at index 1
+Sample input 2:
+Input:"aabbccdd"
+Output: No non repeating character found. All the characters are repeating.
+Time Complexity: O(n*2) - n being the length of string
+Space Complexity: O(n)
+"""
```
Comment (siddharth25pandey) [inline Python/other/left_nonRepeating_char.py]: You are displaying the message **Enter a string:**
But in Sample Input there is no such message.
Please write the same thing as what you are getting after executing the code not only dry code run.

File: Python/README.md [hunk 2]
```diff
@@ -140,11 +140,7 @@ _add list here_
...
-- [Satellite Image Analysis using numpy](other/Satellite_Image_Analysis_using_numpy.ipynb)
-- [Swap Two Nibbles](other/Swap_two_nibbles.py)
-
-- [Unique Number III](other/Unique_number_III.py)
-- [Subsequence check](other/Subsequence_check.py)
-
-- [Subsequence check](other/Subsequence_check.py)
-
+- [Satellite Image analysis using Numpy](other/Satellite%20Image%20Analysis%20using%20numpy.ipynb)
+- [Swap 2 nibbles](other/Swap_two_nibbles.py)
```
Comment (siddharth25pandey) [inline Python/README.md:144]: Why you deleted and added it?

File: Python/README.md [hunk 3]
```diff
@@ -140,11 +140,7 @@ _add list here_
...
+- [Find leftmost non repeating character](other/left_nonRepeating_char.py)
+- [Subsequence check](other/Subsequence_check.py)
```
Comment (siddharth25pandey) [inline Python/README.md:146]: Why you deleted and added it?

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@salonishah01 Do the Suggested changes & tell me the reasons for specific issues
Thanks!

File: Python/other/left_nonRepeating_char.py (hunk 1 above)
Comment (salonishah01) [reply Python/other/left_nonRepeating_char.py]: @siddharth25pandey Specified changes are done.

File: Python/README.md (hunk 2 above)
Comment (salonishah01) [reply Python/README.md:144]: @siddharth25pandey It had showed me error earlier, so had to change it.

**Review by siddharth25pandey — APPROVED** [review]
@salonishah01 Now perfect
Able to merge
Thanks!

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Pull request merged**